src/components/MlTemporalController/MlTemporalController.tsx

```diff
@@ -0,0 +1,355 @@
+	 */
+	labelColor?: string;
+	/**
+	 * Layout property object, that is passed to the labels layer.
+	 * https://maplibre.org/maplibre-gl-js-docs/style-spec/layers/#symbol
+	 */
+	labelLayout?: SymbolLayoutProps;
+	//  | SymbolLayoutArray;
+
+	/**
+	 * Paint property object for the features layer.
+	 * https://maplibre.org/maplibre-gl-js-docs/style-spec/layers/#symbol
+	 */
+	labelPaint?: SymbolPaintProps;
+	/**
+	 * Hover event handler that is executed whenever a geometry rendered by this component is hovered.
+	 */
+	/**
+	 * A numeric value that sets how many steps before the labels start to appear.
+	 * By default it is set to 5 steps.
+	 */
+	labelFadeIn?: number;
+	/**
+	 * A numeric value that sets how many steps the labels fade out after their proper time value.
+	 * By default it is set to 5 steps.
+	 */
+	labelFadeOut?: number;
+	onHover?: MapLayerMouseEvent;
+	/**
+	 * Click event handler that is executed whenever a geometry rendered by this component is clicked.
+	 */
+	onClick?: MapLayerMouseEvent;
+	/**
+	 * Leave event handler that is executed whenever a geometry rendered by this component is
+	 * left/unhovered.
+	 */
+	onLeave?: MapLayerMouseEvent;
+	/**
+	 * Callback function defined by the user to recive the current value in the parent component.
+	 */
+	onStateChange?: React.Dispatch<React.SetStateAction<number | undefined>>;
+}
+
+/**
+ * Select a GeoJSON object to be displayed in a temporal line.
+ *@component
+ */
+
+function getMinVal(geojson: FeatureCollection | undefined, timeField: string) {
+	if (geojson?.features) {
+		let tempFeatures = [...(geojson.features ? geojson.features : [])];
+		tempFeatures.sort((a, b) => (a.properties?.[timeField] < b.properties?.[timeField] ? 1 : -1));
+		return tempFeatures[tempFeatures.length - 1]?.properties?.[timeField] || 0;
+	}
+	return 0;
+}
+
+function getMaxVal(geojson: FeatureCollection | undefined, timeField: string) {
+	if (geojson?.features) {
+		let tempFeatures = [...(geojson?.features ? geojson.features : [])];
+		tempFeatures.sort((a, b) => (a.properties?.[timeField] < b.properties?.[timeField] ? -1 : 1));
+		return tempFeatures[tempFeatures.length - 1]?.properties?.[timeField] || 0;
+	}
+	return 0;
+}
+
+const MlTemporalController = (props: MlTemporalControllerProps) => {
+	const mapHook = useMap({
+		mapId: props.mapId,
+		waitForLayer: props.insertBeforeLayer,
+	});
+	const initializedRef = useRef(false);
+
+	const labelField = props.labelField || props.geojson?.features[0]?.properties?.[0] || '';
+	const minVal = React.useMemo(() => {
+		if (props.minVal) {
+			return props.minVal;
+		}
+		return getMinVal(props.geojson, props.timeField);
+	}, [props.minVal, props.geojson, props.timeField]);
+
+	const maxVal = React.useMemo(() => {
+		if (props.maxVal) {
+			return props.maxVal;
+		}
+		return getMaxVal(props.geojson, props.timeField);
+	}, [props.maxVal, props.geojson, props.timeField]);
+
+	const [type, setType] = useState(props.type || 'circle');
```

Comment:
All useState default props can be added to the defaultProps object in line 351. That way defaults will be listed in the documentation and useState hooks can be initialized using only the props.